### DISCUSSION TITLE
deploy docker image for production k8s or docker-compose deployment

### DISCUSSION THREAD

**Discussion by zironycho** [outline/outline#1321] [General] [answered]
Hi, I am using the Outline wiki nicely, and this is amazing project. 
However docker deployment is some problem I think. This project's docker-compose environment is good for testing and development, but that is not enough to deploy app on the Kubernetes or another production environment.
I think Outline app use very tiny resources(cpu, memory), but below line, you guys already know, it builds the app every time. In that case, it will use lots of resources(cpu, memory). 

https://github.com/outline/outline/blob/efcfda839830767fc92adb0144e8e0f4b8062f9f/Dockerfile#L14

Outline app is already separated environment's variables in dotfile. 
So I would like to suggest, why don't you separate development Dockerfile and [production Dockerfile like this](https://gist.github.com/zironycho/a15b683f82b5177bf1fab5607c78801b).
Production Dockerfile will save the lots of resources. 

Thank you for your efforts.

**Comment by tommoor** [comment, marked as answer]
Yep, there are a few blockers – this is being discussed in several places right now:

https://github.com/outline/outline/discussions/1319

**Reply by zironycho** [reply]
Thank you for answer! It's great news.

**Reply by tommoor** [reply]
As of [v0.45.0](https://github.com/outline/outline/releases/tag/v0.45.0) there is no longer a build step on docker startup
